feature-libs/checkout/base/components/checkout-review/checkout-review-overview/checkout-review-overview.component.ts

```diff
@@ -4,19 +4,68 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  inject,
+} from '@angular/core';
 import { ActiveCartFacade, Cart } from '@spartacus/cart/base/root';
-import { Observable } from 'rxjs';
+import { FeatureConfigService, TranslationService } from '@spartacus/core';
+import { Observable, take } from 'rxjs';
 
 @Component({
   selector: 'cx-checkout-review-overview',
   templateUrl: './checkout-review-overview.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CheckoutReviewOverviewComponent {
+export class CheckoutReviewOverviewComponent implements AfterViewInit {
+  protected document = inject(DOCUMENT, { optional: true });
+  protected translationService = inject(TranslationService, { optional: true });
+  private featureService = inject(FeatureConfigService, { optional: true });
+
   constructor(protected activeCartFacade: ActiveCartFacade) {}
 
+  ngAfterViewInit(): void {
+    this.wrapComponentsWithSectionEl();
+  }
+
   get cart$(): Observable<Cart> {
     return this.activeCartFacade.getActive();
   }
+
+  /**
+   * Wraps checkout review components with section element required
+   * for applying correct a11y practices.
+   */
+  protected wrapComponentsWithSectionEl() {
+    if (
+      this.document &&
+      this.translationService &&
+      this.featureService?.isEnabled('a11yWrapReviewOrderInSection')
+    ) {
+      this.translationService
+        .translate('checkoutReview.reviewOrder')
+        .pipe(take(1))
+        .subscribe((label) => {
+          // We need to delay for a tick to let components render before querying.
+          setTimeout(() => {
+            // These are the components that we need to wrap.
+            const els: any[] = [
+              this.document?.querySelector('cx-checkout-review-payment'),
+              this.document?.querySelector('cx-checkout-review-overview'),
+              this.document?.querySelector('cx-checkout-review-shipping'),
```

Comment:
Maybe we can define
```ts
protected readonly CHECKOUT_COMPONENTS = [
  'cx-checkout-review-payment',
  'cx-checkout-review-overview',
  'cx-checkout-review-shipping',
];
```
And then use it like this
```suggestion
              const elements = this.CHECKOUT_COMPONENTS.map(selector => 
                this.document?.querySelector(selector)
              )
```
That way it would be easier to find and potentially change the selector, add new component or remove one.